agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/common/NetworkFriendlyExceptions.java

```diff
@@ -49,6 +65,36 @@ public static boolean logSpecialOneTimeFriendlyException(
       //  e.g. wrong host address or cannot reach address due to network issues...
       return false;
     }
+
+    IOException ioException = getCausedByOfType(error, IOException.class);
+    if (ioException != null) {
+      if (!alreadySeen.getAndSet(true)) {
```

Comment:
I think setting the `alreadySeen` should happen below if `!foundCiphers` (don't worry about atomicity between get/set)